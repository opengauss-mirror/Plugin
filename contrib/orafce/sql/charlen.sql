--
-- test LPAD family of functions
--

/* cases where one or more arguments are of type CHAR */
SELECT '|' || oracle.lpad('あbcd'::char(8), 10) || '|';
SELECT '|' || oracle.lpad('あbcd'::char(8),  5) || '|';
SELECT '|' || oracle.lpad('あbcd'::char(8), 1) || '|';

SELECT '|' || oracle.lpad('あbcd'::char(5), 10, 'xい'::char(3)) || '|';
SELECT '|' || oracle.lpad('あbcd'::char(5),  5, 'xい'::char(3)) || '|';

SELECT '|' || oracle.lpad('あbcd'::char(5), 10, 'xい'::text) || '|';
SELECT '|' || oracle.lpad('あbcd'::char(5), 10, 'xい'::varchar2(5)) || '|';
SELECT '|' || oracle.lpad('あbcd'::char(5), 10, 'xい'::nvarchar2(3)) || '|';

SELECT '|' || oracle.lpad('あbcd'::text, 10, 'xい'::char(3)) || '|';
SELECT '|' || oracle.lpad('あbcd'::text,  5, 'xい'::char(3)) || '|';

SELECT '|' || oracle.lpad('あbcd'::varchar2(5), 10, 'xい'::char(3)) || '|';
SELECT '|' || oracle.lpad('あbcd'::varchar2(5),  5, 'xい'::char(3)) || '|';

SELECT '|' || oracle.lpad('あbcd'::nvarchar2(5), 10, 'xい'::char(3)) || '|';
SELECT '|' || oracle.lpad('あbcd'::nvarchar2(5),  5, 'xい'::char(3)) || '|';

/* test oracle.lpad(text, int [, text]) */
SELECT '|' || oracle.lpad('あbcd'::text, 10) || '|';
SELECT '|' || oracle.lpad('あbcd'::text,  5) || '|';

SELECT '|' || oracle.lpad('あbcd'::varchar2(10), 10) || '|';
SELECT '|' || oracle.lpad('あbcd'::varchar2(10), 5) || '|';

SELECT '|' || oracle.lpad('あbcd'::nvarchar2(10), 10) || '|';
SELECT '|' || oracle.lpad('あbcd'::nvarchar2(10), 5) || '|';

SELECT '|' || oracle.lpad('あbcd'::text, 10, 'xい'::text) || '|';
SELECT '|' || oracle.lpad('あbcd'::text, 10, 'xい'::varchar2(5)) || '|';
SELECT '|' || oracle.lpad('あbcd'::text, 10, 'xい'::nvarchar2(3)) || '|';

SELECT '|' || oracle.lpad('あbcd'::varchar2(5), 10, 'xい'::text) || '|';
SELECT '|' || oracle.lpad('あbcd'::varchar2(5), 10, 'xい'::varchar2(5)) || '|';
SELECT '|' || oracle.lpad('あbcd'::varchar2(5), 10, 'xい'::nvarchar2(5)) || '|';

SELECT '|' || oracle.lpad('あbcd'::nvarchar2(5), 10, 'xい'::text) || '|';
SELECT '|' || oracle.lpad('あbcd'::nvarchar2(5), 10, 'xい'::varchar2(5)) || '|';
SELECT '|' || oracle.lpad('あbcd'::nvarchar2(5), 10, 'xい'::nvarchar2(5)) || '|';

--
-- test RPAD family of functions
--

/* cases where one or more arguments are of type CHAR */
SELECT '|' || oracle.rpad('あbcd'::char(8), 10) || '|';
SELECT '|' || oracle.rpad('あbcd'::char(8),  5) || '|';
SELECT '|' || oracle.rpad('あbcd'::char(8), 1) || '|';

SELECT '|' || oracle.rpad('あbcd'::char(5), 10, 'xい'::char(3)) || '|';
SELECT '|' || oracle.rpad('あbcd'::char(5),  5, 'xい'::char(3)) || '|';

SELECT '|' || oracle.rpad('あbcd'::char(5), 10, 'xい'::text) || '|';
SELECT '|' || oracle.rpad('あbcd'::char(5), 10, 'xい'::varchar2(5)) || '|';
SELECT '|' || oracle.rpad('あbcd'::char(5), 10, 'xい'::nvarchar2(3)) || '|';

SELECT '|' || oracle.rpad('あbcd'::text, 10, 'xい'::char(3)) || '|';
SELECT '|' || oracle.rpad('あbcd'::text,  5, 'xい'::char(3)) || '|';

SELECT '|' || oracle.rpad('あbcd'::varchar2(5), 10, 'xい'::char(3)) || '|';
SELECT '|' || oracle.rpad('あbcd'::varchar2(5),  5, 'xい'::char(3)) || '|';

SELECT '|' || oracle.rpad('あbcd'::nvarchar2(5), 10, 'xい'::char(3)) || '|';
SELECT '|' || oracle.rpad('あbcd'::nvarchar2(5),  5, 'xい'::char(3)) || '|';

/* test oracle.lpad(text, int [, text]) */
SELECT '|' || oracle.rpad('あbcd'::text, 10) || '|';
SELECT '|' || oracle.rpad('あbcd'::text,  5) || '|';

SELECT '|' || oracle.rpad('あbcd'::varchar2(10), 10) || '|';
SELECT '|' || oracle.rpad('あbcd'::varchar2(10), 5) || '|';

SELECT '|' || oracle.rpad('あbcd'::nvarchar2(10), 10) || '|';
SELECT '|' || oracle.rpad('あbcd'::nvarchar2(10), 5) || '|';

SELECT '|' || oracle.rpad('あbcd'::text, 10, 'xい'::text) || '|';
SELECT '|' || oracle.rpad('あbcd'::text, 10, 'xい'::varchar2(5)) || '|';
SELECT '|' || oracle.rpad('あbcd'::text, 10, 'xい'::nvarchar2(3)) || '|';

SELECT '|' || oracle.rpad('あbcd'::varchar2(5), 10, 'xい'::text) || '|';
SELECT '|' || oracle.rpad('あbcd'::varchar2(5), 10, 'xい'::varchar2(5)) || '|';
SELECT '|' || oracle.rpad('あbcd'::varchar2(5), 10, 'xい'::nvarchar2(5)) || '|';

SELECT '|' || oracle.rpad('あbcd'::nvarchar2(5), 10, 'xい'::text) || '|';
SELECT '|' || oracle.rpad('あbcd'::nvarchar2(5), 10, 'xい'::varchar2(5)) || '|';
SELECT '|' || oracle.rpad('あbcd'::nvarchar2(5), 10, 'xい'::nvarchar2(5)) || '|';

--
-- test oracle.length()
--

/* test that trailing blanks are not ignored */
SELECT oracle.length('あbb'::char(6));
SELECT oracle.length(''::char(6));
