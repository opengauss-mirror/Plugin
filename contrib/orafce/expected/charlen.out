--
-- test LPAD family of functions
--
/* cases where one or more arguments are of type CHAR */
SELECT '|' || oracle.lpad('あbcd'::char(8), 10) || '|';
   ?column?   
--------------
 |   あbcd  |
(1 row)

SELECT '|' || oracle.lpad('あbcd'::char(8),  5) || '|';
 ?column? 
----------
 |あbcd|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::char(8), 1) || '|';
 ?column? 
----------
 | |
(1 row)

SELECT '|' || oracle.lpad('あbcd'::char(5), 10, 'xい'::char(3)) || '|';
   ?column?   
--------------
 |x  x  あbc|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::char(5),  5, 'xい'::char(3)) || '|';
 ?column? 
----------
 |xあbc|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::char(5), 10, 'xい'::text) || '|';
   ?column?   
--------------
 |xいxいあbc|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::char(5), 10, 'xい'::varchar2(5)) || '|';
   ?column?   
--------------
 |xいxいあbc|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::char(5), 10, 'xい'::nvarchar2(3)) || '|';
   ?column?   
--------------
 |xいxいあbc|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::text, 10, 'xい'::char(3)) || '|';
   ?column?   
--------------
 |x  x あbcd|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::text,  5, 'xい'::char(3)) || '|';
 ?column? 
----------
 |あbcd|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::varchar2(5), 10, 'xい'::char(3)) || '|';
   ?column?   
--------------
 |x  x  あbc|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::varchar2(5),  5, 'xい'::char(3)) || '|';
 ?column? 
----------
 |xあbc|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::nvarchar2(5), 10, 'xい'::char(3)) || '|';
   ?column?   
--------------
 |x  x あbcd|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::nvarchar2(5),  5, 'xい'::char(3)) || '|';
 ?column? 
----------
 |あbcd|
(1 row)

/* test oracle.lpad(text, int [, text]) */
SELECT '|' || oracle.lpad('あbcd'::text, 10) || '|';
   ?column?   
--------------
 |     あbcd|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::text,  5) || '|';
 ?column? 
----------
 |あbcd|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::varchar2(10), 10) || '|';
   ?column?   
--------------
 |     あbcd|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::varchar2(10), 5) || '|';
 ?column? 
----------
 |あbcd|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::nvarchar2(10), 10) || '|';
   ?column?   
--------------
 |     あbcd|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::nvarchar2(10), 5) || '|';
 ?column? 
----------
 |あbcd|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::text, 10, 'xい'::text) || '|';
   ?column?   
--------------
 | xいxあbcd|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::text, 10, 'xい'::varchar2(5)) || '|';
   ?column?   
--------------
 | xいxあbcd|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::text, 10, 'xい'::nvarchar2(3)) || '|';
   ?column?   
--------------
 | xいxあbcd|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::varchar2(5), 10, 'xい'::text) || '|';
   ?column?   
--------------
 |xいxいあbc|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::varchar2(5), 10, 'xい'::varchar2(5)) || '|';
   ?column?   
--------------
 |xいxいあbc|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::varchar2(5), 10, 'xい'::nvarchar2(5)) || '|';
   ?column?   
--------------
 |xいxいあbc|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::nvarchar2(5), 10, 'xい'::text) || '|';
   ?column?   
--------------
 | xいxあbcd|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::nvarchar2(5), 10, 'xい'::varchar2(5)) || '|';
   ?column?   
--------------
 | xいxあbcd|
(1 row)

SELECT '|' || oracle.lpad('あbcd'::nvarchar2(5), 10, 'xい'::nvarchar2(5)) || '|';
   ?column?   
--------------
 | xいxあbcd|
(1 row)

--
-- test RPAD family of functions
--
/* cases where one or more arguments are of type CHAR */
SELECT '|' || oracle.rpad('あbcd'::char(8), 10) || '|';
   ?column?   
--------------
 |あbcd     |
(1 row)

SELECT '|' || oracle.rpad('あbcd'::char(8),  5) || '|';
 ?column? 
----------
 |あbcd|
(1 row)

SELECT '|' || oracle.rpad('あbcd'::char(8), 1) || '|';
 ?column? 
----------
 | |
(1 row)

SELECT '|' || oracle.rpad('あbcd'::char(5), 10, 'xい'::char(3)) || '|';
   ?column?   
--------------
 |あbcx  x  |
(1 row)

SELECT '|' || oracle.rpad('あbcd'::char(5),  5, 'xい'::char(3)) || '|';
 ?column? 
----------
 |あbcx|
(1 row)

SELECT '|' || oracle.rpad('あbcd'::char(5), 10, 'xい'::text) || '|';
   ?column?   
--------------
 |あbcxいxい|
(1 row)

SELECT '|' || oracle.rpad('あbcd'::char(5), 10, 'xい'::varchar2(5)) || '|';
   ?column?   
--------------
 |あbcxいxい|
(1 row)

SELECT '|' || oracle.rpad('あbcd'::char(5), 10, 'xい'::nvarchar2(3)) || '|';
   ?column?   
--------------
 |あbcxいxい|
(1 row)

SELECT '|' || oracle.rpad('あbcd'::text, 10, 'xい'::char(3)) || '|';
   ?column?   
--------------
 |あbcdx  x |
(1 row)

SELECT '|' || oracle.rpad('あbcd'::text,  5, 'xい'::char(3)) || '|';
 ?column? 
----------
 |あbcd|
(1 row)

SELECT '|' || oracle.rpad('あbcd'::varchar2(5), 10, 'xい'::char(3)) || '|';
   ?column?   
--------------
 |あbcx  x  |
(1 row)

SELECT '|' || oracle.rpad('あbcd'::varchar2(5),  5, 'xい'::char(3)) || '|';
 ?column? 
----------
 |あbcx|
(1 row)

SELECT '|' || oracle.rpad('あbcd'::nvarchar2(5), 10, 'xい'::char(3)) || '|';
   ?column?   
--------------
 |あbcdx  x |
(1 row)

SELECT '|' || oracle.rpad('あbcd'::nvarchar2(5),  5, 'xい'::char(3)) || '|';
 ?column? 
----------
 |あbcd|
(1 row)

/* test oracle.lpad(text, int [, text]) */
SELECT '|' || oracle.rpad('あbcd'::text, 10) || '|';
   ?column?   
--------------
 |あbcd     |
(1 row)

SELECT '|' || oracle.rpad('あbcd'::text,  5) || '|';
 ?column? 
----------
 |あbcd|
(1 row)

SELECT '|' || oracle.rpad('あbcd'::varchar2(10), 10) || '|';
   ?column?   
--------------
 |あbcd     |
(1 row)

SELECT '|' || oracle.rpad('あbcd'::varchar2(10), 5) || '|';
 ?column? 
----------
 |あbcd|
(1 row)

SELECT '|' || oracle.rpad('あbcd'::nvarchar2(10), 10) || '|';
   ?column?   
--------------
 |あbcd     |
(1 row)

SELECT '|' || oracle.rpad('あbcd'::nvarchar2(10), 5) || '|';
 ?column? 
----------
 |あbcd|
(1 row)

SELECT '|' || oracle.rpad('あbcd'::text, 10, 'xい'::text) || '|';
   ?column?   
--------------
 |あbcdxいx |
(1 row)

SELECT '|' || oracle.rpad('あbcd'::text, 10, 'xい'::varchar2(5)) || '|';
   ?column?   
--------------
 |あbcdxいx |
(1 row)

SELECT '|' || oracle.rpad('あbcd'::text, 10, 'xい'::nvarchar2(3)) || '|';
   ?column?   
--------------
 |あbcdxいx |
(1 row)

SELECT '|' || oracle.rpad('あbcd'::varchar2(5), 10, 'xい'::text) || '|';
   ?column?   
--------------
 |あbcxいxい|
(1 row)

SELECT '|' || oracle.rpad('あbcd'::varchar2(5), 10, 'xい'::varchar2(5)) || '|';
   ?column?   
--------------
 |あbcxいxい|
(1 row)

SELECT '|' || oracle.rpad('あbcd'::varchar2(5), 10, 'xい'::nvarchar2(5)) || '|';
   ?column?   
--------------
 |あbcxいxい|
(1 row)

SELECT '|' || oracle.rpad('あbcd'::nvarchar2(5), 10, 'xい'::text) || '|';
   ?column?   
--------------
 |あbcdxいx |
(1 row)

SELECT '|' || oracle.rpad('あbcd'::nvarchar2(5), 10, 'xい'::varchar2(5)) || '|';
   ?column?   
--------------
 |あbcdxいx |
(1 row)

SELECT '|' || oracle.rpad('あbcd'::nvarchar2(5), 10, 'xい'::nvarchar2(5)) || '|';
   ?column?   
--------------
 |あbcdxいx |
(1 row)

--
-- test oracle.length()
--
/* test that trailing blanks are not ignored */
SELECT oracle.length('あbb'::char(6));
 length 
--------
      4
(1 row)

SELECT oracle.length(''::char(6));
 length 
--------
       
(1 row)

